src/api/check.rs

```diff
@@ -963,7 +953,7 @@ impl Response {
 #[derive(Debug, Clone, PartialEq)]
 pub struct ResponseWithContext {
     /// Original text that was checked by LT.
-    pub text: Cow<'static, str>,
+    pub text: String,
```

Comment:
Why does it have to be an owned string? If I am correct, the "context" is simply a source text, so should be `&'source str`, no?

Reply:
I just had issues with lifetimes, in particular `check_multiple_and_join` in `server.rs` was a pain due to those tokio tasks that are spawned. However, I'm having a look at if it can maybe work with `IntoStatic`.

What is the match positions section in this code block supposed to do? It's causing a lifetime issue and I don't see the values being returned / used for anything:

```rust
impl<'source> From<ResponseWithContext<'source>> for Response {
    #[allow(clippy::needless_borrow)]
    fn from(mut resp: ResponseWithContext<'source>) -> Self {
        let iter: MatchPositions<'_, std::slice::IterMut<'_, Match>> = (&mut resp).into();

        for (line_number, line_offset, m) in iter {
            m.more_context = Some(MoreContext {
                line_number,
                line_offset,
            });
        }
        resp.response
    }
}
```

Reply:
Hum I see, probably using multiple tasks may conflict with lifetime.

The match `iter` is an iterator that will automatically count lines and line offsets, so we can then add this information to the `Match` object.

Reply:
Thanks that helped, so that piece of code modifies the response with some additional context before returning it.

Have a look at the latest commit and see if you're happy with the changes as I needed some workarounds for more lifetime issues due to that block of code above.

I could improve it if we drop the implementation for `Iter` and just keep the one for `IterMut`, so that it doesn't need to be generic. Also, maybe the modification of `response` with that additional context could happen when `ResponseWithContext` is first created, rather than only when converting it to `Response`?

Reply:
Hi @Rolv-Apneseth! Your work looks very promising, but I have a paper deadline (next Thursday) that is taking most of my time, so I will (hopefully, if the deadline is not postponed) have time to review your work on next Friday :-)

Reply:
Of course, no worries, you can take your time. And good luck!

Reply:
Were you alright with these changes then? If so I can move on to the de-duplication of the `DataAnnotation` and `CliRequest` methods

Reply:
Also - hope the paper went well